pkg/services/alerting/notifiers/slack.go

```diff
@@ -174,9 +190,20 @@ func (sn *SlackNotifier) Notify(evalContext *alerting.EvalContext) error {
 		})
 	}
 
-	message := sn.Mention
+	message := ""
+	mentionGroup := strings.TrimSpace(sn.MentionGroup)
+	if mentionGroup != "" {
+		message += fmt.Sprintf("<!subteam^%s>", mentionGroup)
+	}
+	mentionUser := strings.TrimSpace(sn.MentionUser)
+	if mentionUser != "" {
+		message += fmt.Sprintf("<@%s>", mentionUser)
```

Comment:
Previously we [assumed](https://grafana.com/docs/grafana/latest/alerting/notifications/#slack) that `@` was included in the user input. I think that it will be confusing and prone to errors if we make the users change usernames to IDs and at the same time remove the `@` character from their existing notifiers.

In any case the existing documentation should be updated too and probably we need to add a note to the next release for letting the users fix their notifiers.

Reply:
@papagian My thinking at least is that this is such a radical change, splitting user and group mentions (whereas they used to be one). Also, the `@` sign only works for user mentions now, since the character technically happens to be part of the notification syntax. It doesn't work for group mentions, due to its different syntax, so isn't it just more confusing if you have to use `@` for user mentions and not for group mentions?

Reply:
I haven't thought about other types of mentions though, than user or group, since the old tooltip just describes those types of mentions. Not sure how many types of mention targets we should support.

Reply:
probably yes! so just make sure we communicate this properly to the users.